optuna/trial/_frozen.py

```diff
@@ -199,3 +199,95 @@ def duration(self):
             return self.datetime_complete - self.datetime_start
         else:
             return None
+
+
+@experimental("2.0.0")
+def create_trial(
```

Comment:
I prefer more concrete function name like `create_frozen_trial()` because someone might add `create_fixed_trial()` or something similar function in the future.

Reply:
I agree with you in terms of a different viewpoint. The names of `optuna.create_trial` and `optuna.create_study` seem to be symmetrical, but `optuna.create_trial` is rarely used compared with `optuna.create_study`. So, I feel `optuna.create_trial` might be too general, and I also prefer the name like `create_frozen_trial()`.

Reply:
I may be too anxious, but some users may have trouble when they use `create_trial` and `add_trial` as an alternative to `study.optimize`. 

After emerging this PR, we can write an iteration of `study.optimize` using public API as follows:

```python
trial = optuna.create_trial( ... )
trial.value = objective(trial)
study.add_trial(trial)
```

This approach is more flexible than `study.optimize` but users need to take care of exceptions including `TrialPruned` and the trial state management by themselves. So, it may have many potholes and I think we cannot recommend to novice users.

I think providing the `optuna.create_trial` and `Study.add_trial` provides many merits to users, but we may carefully guide the users.

The followings are just examples:

- Provide `create_trial` as a function of `optuna.trial` module: `optuna.trial.create_trial()`
- Add tutorial for `create_trial` and `add_trial`

Reply:
Thanks for the important feedback. It's now explicitly mentioned that this is not just experimental but also a low-level API. And since it's not mirroring `create_study`, it's not moved to `optuna.trial` instead of `optuna`.

As for the name, it's kept as is. I believe we can be more specific of what type of trial we are talking about, when we really need to distinguish them. In current Optuna, we usually don't in the user facing API (e.g. "get_trial").